### PR TITLE
Feat: add union type to nnc field

### PR DESCRIPTION
**Pull request by avivash**
# Description

Updating the `nnc` schema field to support objects of IPLD bytes. This is needed so we can send a nonce to Homestar to enable replays

**Note: I haven't tested this yet, but it looks like `zod` supports union types in this format 🤷🏼**

## Link to sister PR

https://github.com/everywhere-computer/every-cli/pull/6

## Type of change

- [x] Refactor (non-breaking change that updates existing functionality)

### CODE REVIEW

File: packages/homestar/src/schemas.js [hunk 1]
```diff
@@ -62,7 +62,10 @@ export const Receipt = z.object({
 export const Task = z.object({
   op: z.string(),
   rsc: z.string().url(),
-  nnc: z.string(),
+  nnc: z.union([
+    z.string(),
```
Comment (hugomrdias) [inline packages/homestar/src/schemas.js:66]: this should be empty string, not any string according to homestar schema

**Review by hugomrdias — CHANGES_REQUESTED** [review]

File: packages/homestar/src/schemas.js (hunk 1 above)
Comment (avivash) [reply packages/homestar/src/schemas.js]: @hugomrdias thanks 👍🏼 is `z.string().length(0)` the correct way to do that with `zod`? That's what I'm assuming based on the [docs](https://zod.dev/?id=strings)

Comment (hugomrdias) [reply packages/homestar/src/schemas.js]: yep

**Review by hugomrdias — APPROVED** [review]

**Comment by avivash** [comment]
i borked the signing. i'll clean this up shortly

**Pull request merged**